celements-model/src/main/java/com/celements/store/part/CelHibernateStoreDocumentPart.java

```diff
@@ -92,34 +98,28 @@ public void saveXWikiDoc(XWikiDocument doc, XWikiContext context, boolean bTrans
               doc.getDocumentArchive(context);
             }
           } catch (XWikiException xwe) {
-            LOGGER.debug("saveXWikiDoc - this is a non critical error: {} {}", doc.getId(),
-                store.getModelUtils().serializeRef(doc.getDocumentReference()), xwe);
+            LOGGER.debug("saveXWikiDoc - this is a non critical error: {} {}",
+                doc.getId(), defer(() -> store.serialize(doc, GLOBAL)), xwe);
           }
         }
       }
 
-      DocumentReference existingDocRef = checkExistingDoc(doc, session);
-      if (existingDocRef == null) {
+      /*
+       * FIXME [CELDEV-248] ModelLockService
+       * the computed id needs to be locked until the save has happened to prevent overrides.
+       */
+      Session session = savePrepCmd.getSession();
+      if (doc.isNew()) {
         session.save(doc);
-      } else if (existingDocRef.equals(doc.getDocumentReference())) {
```

Comment:
I ask to keep collision detection as long as we use partial hash values as doc ids.

Reply:
this check has moved to `DocumentSavePreparationCommand#checkForCollisions`